tests/fixtures/body-parameter/expected/Resource/TestResource.php

```diff
@@ -17,7 +17,7 @@ class TestResource extends Resource
     public function testSimpleBodyParameter($testString, $parameters = [], $fetch = self::FETCH_OBJECT)
     {
         $queryParam = new QueryParam();
-        $url        = '/test-simple';
+        $url        = 'http://localhost/test-simple';
         $url        = $url . ('?' . $queryParam->buildQueryString($parameters));
         $headers    = array_merge(['Host' => 'localhost'], $queryParam->buildHeaders($parameters));
```

Comment:
On some projects we modify the Host header with a plugin, because depending on the env, we need to hit different hosts. I fear that your PR will make this use case harder because the URL is now full. Am I wrong?